### PR TITLE
mirror: allow updating blacklisted ids

### PR DESCRIPTION
**Pull request by decentralion**
Now the GitHub mirror will warn when the blacklisted ids are
inconsistent, rather than throwing an error.

This bit of tolerance is quite convenient, because you might know that
the change in blacklisted ids was for a totally different repo, and have
a reason to want to keep your cache around. Of course, you'll eventually
want to reset the cache so that you stop seeing the warning message. :)

Test plan: Tests updated; `yarn test` passes.

### CODE REVIEW

**Comment by wchargin** [comment]
What unexpected behavior may result? Data loss? Incorrect responses?
Failure to terminate?

File: src/graphql/mirror.js [hunk 1]
```diff
@@ -200,16 +207,36 @@ export class Mirror {
...
+            "Database already populated with incompatible schema."
+          );
+        } else if (
+          !deepEqual(existingConfig.blacklistedIds, mirrorConfig.blacklistedIds)
+        ) {
+          console.error(dedent`\
+            Warning: Database already loaded with different blacklistedIds.\
+            Unexpected behavior may result. Consider resetting your cache.`);
+          // We don't need to do the rest of the initialization, because it was
+          // setup, albiet with different options.
```
Comment (wchargin) [inline src/graphql/mirror.js]: sp.: *albeit* (“although be it”)

Also, *set up* as a phrasal verb is two words.

File: src/graphql/mirror.js [hunk 2]
```diff
@@ -200,16 +207,36 @@ export class Mirror {
...
+            "Database already populated with incompatible version."
+          );
+        } else if (!deepEqual(existingConfig.schema, mirrorConfig.schema)) {
+          throw new Error(
+            "Database already populated with incompatible schema."
+          );
+        } else if (
+          !deepEqual(existingConfig.blacklistedIds, mirrorConfig.blacklistedIds)
+        ) {
+          console.error(dedent`\
```
Comment (wchargin) [inline src/graphql/mirror.js]: A warning should use `console.warn`, no?

File: src/graphql/mirror.js [hunk 3]
```diff
@@ -200,16 +207,36 @@ export class Mirror {
...
+            Warning: Database already loaded with different blacklistedIds.\
```
Comment (wchargin) [inline src/graphql/mirror.js]: Also, this won’t format as you expect; it will have 12 interior spaces.

**Comment by decentralion** [comment]
I actually have no idea what kinds of bad things may happen. They may happen, and they are truly unexpected. :)

**Comment by wchargin** [comment]
FWIW, the fact that this PR was not merged just saved me a bit of
trouble. I’d run

```
SOURCECRED_DIRECTORY=/tmp/testsc/ node ./bin/sourcecred.js load aragon/aragon
```

to test some behavior on a clean cache. I went off and did some
investigation, then came back and wanted to retry the load from where it
failed, so I reran my last `node ./bin/sourcecred.js` invocation—but
this omitted the environment variable, keying against my primary cache
instead. Happily, the load pipeline failed immediately with

> Error: Database already populated with incompatible schema, options,
> or version

rather than corrupting the database that had the proper set of
blacklisted IDs.

**Comment by decentralion** [comment]
Closing as I don't feel we have consensus this is a good idea.

**Pull request closed without merging**